android/app/src/main/java/com/rjnr/pocketnode/data/wallet/KeyManager.kt

```
@@ -159,6 +159,38 @@ class KeyManager @Inject constructor(
+     * holds keys (`total > 0`), exactly mirroring [hasWallet]'s short-circuit.
+     * Touching `prefs` lazily initializes `encryptedPrefs`, which can flip
+     * `walletResetDueToCorruption` on an unreadable legacy file; gating on
+     * `total > 0` keeps a migrated wallet's routing untouched (Codex/CodeRabbit
+     * review on #425). Each probe is independently guarded so one failure still
+     * yields a usable line rather than dropping the whole log event.
+     */
+    suspend fun diagnosticKeyState(): String {
+        val helper = keyStoreMigrationHelper
+        val (total, v1, v2) = try {
+            helper?.keyMaterialCounts() ?: Triple(-1, -1, -1)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)